File: BraintreeCore/src/test/java/com/braintreepayments/api/GraphQLConfigurationUnitTest.java
package com.braintreepayments.api;

import com.braintreepayments.api.GraphQLConstants.Features;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertTrue;
import static org.junit.Assert.assertFalse;

@RunWith(RobolectricTestRunner.class)
public class GraphQLConfigurationUnitTest {

    @Test
    public void fromJson_parsesFullInput() throws JSONException {
        JSONObject input = new JSONObject()
                .put("url", "https://example.com/graphql")
                .put("features", new JSONArray()
                        .put("tokenize_credit_cards")
                );
        GraphQLConfiguration sut = GraphQLConfiguration.fromJson(input);
        assertTrue(sut.isEnabled());
        assertTrue(sut.isFeatureEnabled(Features.TOKENIZE_CREDIT_CARDS));
        assertEquals("https://example.com/graphql", sut.getUrl());
    }

    @Test
    public void fromJson_whenInputNull_returnsConfigWithDefaultValues() {
       GraphQLConfiguration sut = GraphQLConfiguration.fromJson(null);
       assertFalse(sut.isEnabled());
       assertFalse(sut.isFeatureEnabled(Features.TOKENIZE_CREDIT_CARDS));
       assertEquals("", sut.getUrl());
    }

    @Test
    public void fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        GraphQLConfiguration sut = GraphQLConfiguration.fromJson(new JSONObject());
        assertFalse(sut.isEnabled());
        assertFalse(sut.isFeatureEnabled(Features.TOKENIZE_CREDIT_CARDS));
        assertEquals("", sut.getUrl());
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/GraphQLConfigurationUnitTest.kt
package com.braintreepayments.api

import org.robolectric.RobolectricTestRunner
import org.json.JSONArray
import org.json.JSONObject
import org.junit.Assert.*
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class GraphQLConfigurationUnitTest {
    @Test
    fun fromJson_parsesFullInput() {
        val input = JSONObject()
            .put("url", "https://example.com/graphql")
            .put(
                "features", JSONArray()
                    .put("tokenize_credit_cards")
            )
        val sut = GraphQLConfiguration(input)
        assertTrue(sut.isEnabled)
        assertTrue(sut.isFeatureEnabled(GraphQLConstants.Features.TOKENIZE_CREDIT_CARDS))
        assertEquals("https://example.com/graphql", sut.url)
    }

    @Test
    fun fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        val sut = GraphQLConfiguration(null)
        assertFalse(sut.isEnabled)
        assertFalse(sut.isFeatureEnabled(GraphQLConstants.Features.TOKENIZE_CREDIT_CARDS))
        assertEquals("", sut.url)
    }

    @Test
    fun fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        val sut = GraphQLConfiguration(JSONObject())
        assertFalse(sut.isEnabled)
        assertFalse(sut.isFeatureEnabled(GraphQLConstants.Features.TOKENIZE_CREDIT_CARDS))
        assertEquals("", sut.url)
    }
}
